src/greenlight/step.clj

```diff
@@ -95,12 +95,14 @@
   (create-foo
     {:foo/name \"Foo 1\"}
     :output :foo.1/id)"
-  [step-name docstring & default-config]
-  (when-not (string? docstring)
-    (throw (IllegalArgumentException.
-             "defstep expects a docstring for the step constructor")))
-  (let [default-config (qualify-config-keys
-                         (apply hash-map default-config))]
+  [step-name & step-decl]
+  (let [docstring (if (string? (first step-decl)) (first step-decl) "")
```

Comment:
Can you update this to only include the `docstring` on the step-name function when the value is provided, instead of defaulting to an empty string?